MAINTAINERS.md

```diff
@@ -26,9 +26,24 @@ Envoy internals.
-
-- A change to [repositories.bzl](bazel/repositories.bzl) to update the commit and SHA.
-- A sync of [.bazelrc](.bazelrc) with [Envoy's version](https://github.com/envoyproxy/envoy/blob/main/.bazelrc) to update our build configurations.
-- A sync of the build image sha used in the [ci configuration](.circleci/config.yml) with [Envoy's version](https://github.com/envoyproxy/envoy/blob/main/.circleci/config.yml) to sync our CI testing environment.
-- Sometimes the dependency update comes with changes that break our build. We include any changes required to Nighthawk to fix that.
+We aim to [synchronize our Envoy dependency](https://github.com/envoyproxy/nighthawk/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+%22update+envoy%22+) with the latest revision weekly. Nighthawk reuses large parts of Envoy's build system and codebase, so keeping Nighthawk up to date with Envoy's changes is an important maintenance task. When performing the update, follow this procedure:
+
+1. Create a fork of Nighthawk, or fetch upstream and merge changes into your fork if you already have one.
+1. Create a new branch from `main`, e.g. `envoy-update`.
+1. Edit [bazel/repositories.bzl](bazel/repositories.bzl).
+   1. Update `ENVOY_COMMIT` to the latest Envoy's commit from [this page](https://github.com/envoyproxy/envoy/commits/main). (Clicking on the short commit id opens a page that contains the fully expanded commit id).
+   1. Set `ENVOY_SHA` to an empty string initially, we will get the correct sha256 after the first bazel execution.
+   Example content of `bazel/repositories.bzl` after the edits:
+   ```
+   ENVOY_COMMIT = "9753819331d1547c4b8294546a6461a3777958f5"  # Jan 24th, 2021
+   ENVOY_SHA = ""
+   ```
+  1. Run `ci/do_ci.sh build`, notice the sha256 value at the top of the output, example:
+  ```
+  INFO: SHA256 (https://github.com/envoyproxy/envoy/archive/9753819331d1547c4b8294546a6461a3777958f5.tar.gz) = f4d26c7e78c0a478d959ea8bc877f260d4658a8b44e294e3a400f20ad44d41a3
+  ```
+  1. Update `ENVOY_SHA` in [bazel/repositories.bzl](bazel/repositories.bzl) to this value.
+1. Sync (copy) [.bazelrc](.bazelrc) from [Envoy's version](https://github.com/envoyproxy/envoy/blob/main/.bazelrc) to update our build configurations. Be sure to retain our local modifications, all lines that are unique to Nighthawk are marked with comment `# unique`.
+1. Sync (copy) [.bazelversion](.bazelversion) from [Envoy's version](https://github.com/envoyproxy/envoy/blob/main/.bazelversion) to ensure we are using the same build system version.
+1. Run `ci/do_ci.sh test`. Sometimes the dependency update comes with changes that break our build. Include any changes required to Nighthawk to fix that in the same PR.
+1. Create a PR with title like `Update Envoy to 9753819 (Jan 24th 2021)`, describe all performed changes in the PR's descriotion.
```

Comment:
nite: "with a title like"

Reply:
- We might also mention that sometimes an Envoy update may come with CLI arg changes. Then regenerating the readme files is needed: `tools/update_cli_readme_documentation.sh --mode fix` to avoid failure in CI.
- The same could happen to formatting, we may pull in new formatting requirements. Then `ci/do_ci.sh fix_format` ought to auto update to the new situation.

Reply:
Good idea, done in 4f4ff1cc6061ff981e95470759cd80db561ca81e.